object_store_factory/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+mod aws;
+mod google;
+mod local;
+mod memory;
+
+use object_store::parse_url_opts;
+use object_store::ObjectStore;
+use object_store::ObjectStoreScheme;
+use std::collections::HashMap;
+use tracing::warn;
+use url::Url;
+
+/// Configuration for object storage, holding settings in a key-value format.
+pub struct Config {
+    pub settings: HashMap<String, String>,
+}
```

Comment:
Looks like this isn't being used anywhere?

On the other hand if it is, you can declare it as a type alias, since it's a simple wrapper
```rust
pub type StorageConfig = HashMap<String, String>;
```

Reply:
I declared it as type alias as sugested. This is used at `build_object_store_from_config`.